include/boost/json/value.hpp

```diff
@@ -3293,6 +3293,111 @@ class value
     find_pointer(string_view ptr, std::error_code& ec) noexcept;
 /** @} */
 
+    //------------------------------------------------------
+
+    /** Set an element via JSON Pointer.
+
+        This function is used to insert or assign to a potentially nested
+        element of the value using a JSON Pointer string. The function may
+        create intermediate elements corresponding to pointer segments.
+        <br/>
+
+        For a traversed pointer segment, if the corresponding element does
+        not exist or is a `null` value
+
+        @li if the containing value is an @ref object, a new
+          @ref key_value_pair is inserted with the segment's _unescaped_ string
+          as key and empty object as value; otherwise
+
+        @li if the containing value is an @ref array, then if the segment is
+          pass-the-end marker (`-`) or if it is a number equal to array's
+          size, then an empty object is appended; if the segment is larger
+          than the array's size, the function produces an error; otherwise
+
+        @li if the containing value is a `null` value, then it is replaced with
+          an empty object and considered by the function again.
+        <br/>
+
+        The type of created intermediate elements is always @ref object. The
+        function creates at most one element for each pointer segment.
+
+        @par Complexity
+        Linear in the size of `ptr`.
+
+        @par Exception Safety
+        Basic guarantee.
```

Comment:
but we could offer the Strong Guarantee, no?  If we do two passes through the pointer?